src/widgets/buttons.ts

```diff
@@ -478,6 +488,42 @@ export function deleteButtonWithCheck (
   return deleteButtonElt
 }
 
+function getButtonStyle (options: ButtonWidgetOptions = { buttonType: ButtonType.Primary, filled: true }) {
+  // We need to accomadate for legacy code, which is why we have to allow buttonType and filled to be optional
+  let backgroundColor: (ButtonType | string | undefined) = options.buttonType
+  let fontColor: (ButtonType | string | undefined) = '#ffffff'
+  let borderColor: (ButtonType | string | undefined) = options.buttonType
+  let hoverBackgroundColor: (ButtonType | string | undefined) = `lighten(${options.buttonType}, 5%)`
+  let hoverFontColor: (ButtonType | string | undefined) = fontColor
+  if (!options.filled) {
+    backgroundColor = '#ffffff'
+    fontColor = options.buttonType
+    borderColor = options.buttonType
+    hoverBackgroundColor = options.buttonType
+    hoverFontColor = backgroundColor
```

Comment:
This is confusing to read, you have five colours, but two are strings and three are types.

Reply:
yeah i could change to strings.. i didn't like it either, but there were so many variables that need to get set depending on what is selected.